src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -925,7 +949,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                             allowOmissionOfConditionalCalls:=_moduleBeingBuiltOpt.AllowOmissionOfConditionalCalls,
                             isBodySynthesized:=True)
 
-                        If Not diagnosticsThisMethod.HasAnyErrors Then
+                        If Not diagnosticsThisMethod.HasAnyErrors AndAlso DoEmitPhase Then
```

Comment:
>DoEmitPhase [](start = 74, length = 11)

nit: consider swapping checks (I assume `diagnosticsThisMethod.HasAnyErrors` is more expensive). Possibly also below